### PR TITLE
feat: add redis

### PR DESCRIPTION
**Pull request by BaileyMillerSSI**
### Description

Adds Redis as a way to keep track of stock. Redis connection string can be passed in to meet many different configurations.

### Testing

1. Startup redis (I used docker) with the default port
2. Setup your .env file to set the url (REDIS_URL="redis://localhost:6379/1")
3. Run the test command and verify that redis has a key in it.

### New dependencies
`redis`
`@types/redis`

### CODE REVIEW

**Comment by jef** [comment]
Great! This will be useful for things coming up in 4.0.0 as well.

Thank you!

**Comment by BaileyMillerSSI** [comment]
@jef Of course! I am fixing these CI errors and I am going to add more options to pass into the redis client.

**Comment by BaileyMillerSSI** [comment]
@jef 
I am having trouble getting the eslint step to pass. Are you able to provide any feedback on that? I cannot seem to get eslint to auto fix the problems so I am not sure how to fix some of the errors.

**Comment by jef** [comment]
> @jef
> I am having trouble getting the eslint step to pass. Are you able to provide any feedback on that? I cannot seem to get eslint to auto fix the problems so I am not sure how to fix some of the errors.

Haha, I'm sorry. I can be hell sometimes. Let's take a look when it's done building.

**Comment by BaileyMillerSSI** [comment]
> > @jef
> > I am having trouble getting the eslint step to pass. Are you able to provide any feedback on that? I cannot seem to get eslint to auto fix the problems so I am not sure how to fix some of the errors.
> 
> Haha, I'm sorry. I can be hell sometimes. Let's take a look when it's done building.

I think I got it this time, I am using the prettier extension for VS Code but some of the auto formats are against the CI rules but I think I managed to figure it out through trial and error.

File: docs/reference/notification.md [hunk 1]
```diff
@@ -191,3 +191,10 @@ Generate all Twitter keys at: https://developer.twitter.com/
 | `TWITCH_ACCESS_TOKEN` | Twitch access token |
 | `TWITCH_REFRESH_TOKEN` | Twitch refresh token |
 | `TWITCH_CHANNEL` | Twitch channel |
+
+## Redis
+
+| Environment variable | Description |
+|:---:|---|
+| `REDIS_HOST` | IP Address of instance |
```
Comment (BaileyMillerSSI) [inline docs/reference/notification.md:199]: I need to update this.

**Comment by BaileyMillerSSI** [comment]
@jef 
I think I am done implementing this, please feel free to review this whenever you get a chance. 

I am currently letting this run for a bit so I can see how it looks once something comes into stock.

**Review by jef — DISMISSED** [review]
Awesome! I love it. We should add this to the docker-compose sometime. Could be cool.

Thanks!

**Review by jef — APPROVED** [review]

**Comment by gigi2006** [comment]
Windows User Need redis Server start
I have fix this for me Whit this https://github.com/microsoftarchive/redis/releases/latest

**Comment by jef** [comment]
Made a fix in the latest commits to not have it required.

Thank you!

**Pull request merged**